### ISSUE TITLE
Move to new file not working using FZF Lua 

### ISSUE THREAD

**Issue by marcoSven** [pmizio/typescript-tools.nvim#231]
"Move to new file" not working with typescript-tools and [FZF Lua](https://github.com/ibhagwan/fzf-lua) Code actions.

```lua 
require("typescript-tools").setup {
  settings = {
      tsserver_logs = "verbose",
      code_lens_mode = "all",
      tsserver_file_preferences = {
        includeCompletionsForModuleExports = true,
        includeInlayParameterNameHints = "all",
        includeInlayParameterNameHintsWhenArgumentMatchesName = true,
        includeInlayFunctionParameterTypeHints = true,
        includeInlayVariableTypeHints = true,
        includeInlayPropertyDeclarationTypeHints = true,
        includeInlayFunctionLikeReturnTypeHints = true,
        includeInlayEnumMemberValueHints = true,
        importModuleSpecifierPreference = "non-relative",
        quotePreference = "auto"
      },
      separate_diagnostic_server = true,
      publish_diagnostic_on = "insert_leave",
      expose_as_code_action = "all",
      tsserver_max_memory = "auto",
      complete_function_calls = true,
      jsx_close_tag = {
        enable = true,
        filetypes = {"javascriptreact", "typescriptreact"}
      }
    }

```

"Move to new file" is working with lspconfig and FZF Lua

```lua
lspconfig["tsserver"].setup({})

```

This is a similar issue https://github.com/ibhagwan/fzf-lua/issues/1007 

To reproduce the issue with typescript-tools

`require('fzf-lua').setup({'fzf-native'})`

The issue can be resolved by using the tmux profile "fzf-tmux" please see [comment](https://github.com/ibhagwan/fzf-lua/issues/1007#issuecomment-1901717459)

**Comment by yavorski** [comment]
Maybe this is related too did not know where to properly report it ... https://github.com/ibhagwan/fzf-lua/issues/949 https://github.com/ibhagwan/fzf-lua/issues/949#issuecomment-1972838951

**Comment by yavorski** [comment]
I believe this comment from @ibhagwan will be useful here. Please check it out https://github.com/ibhagwan/fzf-lua/issues/949#issuecomment-1973624200 

---

> However, the cause of this error is not fzf-lua but `typescript-tools.nvim` 
> In order to display the diff for the code action preview, if the action doesn't contain the workspace edit section, fzf-lua will send a `codeAction/resolve` request to the LSP server, the first call works (as you found out) but any subsequent call is bugged (and returns `nil`) with `typescript-tools.nvim`.
If fzf-lua sends a synchrounous request with:
```lua
:FzfLua lsp_code_actions previewer=codeaction_native
```
> `typescript-tools.nvim` returns a nil action, with the latest commit, fzf-lua will fallback to the original unresolved action and display this message (no error but also no diff):

![image](https://github.com/ibhagwan/fzf-lua/assets/59988195/cf9b98b6-6985-4427-9ffd-05d73cabf702)

> If using the neovim builtin (non-native) previewer, fzf-lau will send an async LSP request and waits for a callback which never returns, e.g.:
```lua
:FzfLua lsp_code_actions previewer=codeaction
```
> Resulting in:
![image](https://github.com/ibhagwan/fzf-lua/assets/59988195/9e12af01-759b-4667-8b9c-d9d913c78a07)


> Hopefully this is enough information for you to open an upstream issue with `typescript-tools.nvim`.

_Originally posted by @ibhagwan in https://github.com/ibhagwan/fzf-lua/issues/949#issuecomment-1973624200_

**Comment by yavorski** [comment]
I believe this https://github.com/ibhagwan/fzf-lua/issues/949#issuecomment-1973624200 from @ibhagwan about related issue would be useful here:

> In order to display the diff for the code action preview, if the action doesn't contain the workspace edit section, `fzf-lua` will send a `codeAction/resolve` request to the LSP server, the first call works (as you found out) but any subsequent call is bugged (and returns `nil`) with `typescript-tools.nvim`.
If `fzf-lua` sends a synchrounous request with:

```lua
:FzfLua lsp_code_actions previewer=codeaction_native
```

> `typescript-tools.nvim` returns a nil action, with the latest commit, `fzf-lua` will fallback to the original unresolved action and display this message (no error but also no diff):
![image](https://github.com/ibhagwan/fzf-lua/assets/59988195/cf9b98b6-6985-4427-9ffd-05d73cabf702)

>If using the neovim builtin (non-native) previewer, `fzf-lua` will send an async LSP request and waits for a callback which never returns, e.g.:
```lua
:FzfLua lsp_code_actions previewer=codeaction
```

>Resulting in:
![image](https://github.com/ibhagwan/fzf-lua/assets/59988195/9e12af01-759b-4667-8b9c-d9d913c78a07)

_Originally posted by @ibhagwan in https://github.com/ibhagwan/fzf-lua/issues/949#issuecomment-1973624200_

**Comment by ibhagwan** [comment]
> I use telescope so I previously didn't bump into this problem. I'll look into it

@pmizio, the reason you didn’t bump into is probably because this requires using code action previewes, I believe Telescope doesn’t come bulletin with code action previews unless you’re using:
https://github.com/aznhe21/actions-preview.nvim

Based on my testing this issue only happens after sending a `codeAction/resolve` request which doesn’t get called unless you’re using fzf-lua **with preview** or using the above Telescope extension.

**Comment by ibhagwan** [comment]
Oh and @marcoSven, I’m pretty sure the "Move to new file" will work if you disable the code actions previewer by setting `lsp.code_actions.previewer=false` at `setup`.

**Comment by marcoSven** [comment]
@ibhagwan yes that is what I did, without preview "Move to new file" is working.
